### DISCUSSION TITLE
Quasar 2.6.0 Router index file not getting parameters?

### DISCUSSION THREAD

**Discussion by piperone** [quasarframework/quasar#12827] [General - Components / Directives / etc] [answered]
After upgrading to Quasar 2.6.0 today, I'm no longer receiving the parameter in `src/router/index.js`'s default export. Out of the box, the line looks like this:

```
export default route(function (/* { store, ssrContext } */) { ...
```

I uncommented the parameters, so I should be able to expect the store here, right? I'm just getting an empty object, however. This was not the case in 2.5.5. My app relies on being able to access the store in this function, so I had to roll back the upgrade.

Am I looking at a bug here, or do I need to do something differently from before?

Thanks.

**Comment by rstoenescu** [comment, marked as answer]
This is a functionality of the Quasar CLI, not Quasar UI. But you haven't specified which. A reproduction repo would help in uncovering what's going on.

**Reply by piperone** [reply]
Yes, you're right. My apologies. I saw "General" and just chose that. Perhaps there should be a "CLI SPA-mode" category though, because my question doesn't relate to any of the existing ones (BEX, SSR, PWA, Capacitor or Electron). Should I move my question to one of the others?

I don't think a reproduction repo is necessary, because the issue is evident in out-of-the-box Quasar apps. Here's how to reproduce:

- Install via `yarn create quasar` - Choose the 'CLI with Webpack'-variant
- Once created, change the line I referenced above in `src/router/index.js`:
```javascript
export default route(function (parameter) {
    console.log(parameter)
(...)
}
```
- Run the app with `quasar dev`
- Output in console is `Object { }` - an empty object. The store should be available here (and, I assume, the ssrContext in SSR-mode)

This was originally just a question, but if this is a bug, I'd be happy to make an issue for it. It could possibly related to [this](https://github.com/quasarframework/quasar/issues/12793), but I'm not sure.

Update:
Also tested with CLI and Vite. Same issue is present. The result is also the same with both Vuex and Pinia chosen as Store provider.

**Reply by rstoenescu** [reply]
It's working just fine... I double checked again just now.

In fact, I recently enhanced Pinia support recently and should this have been an issue I would have immediately see it. But regardless of having a store or not, `parameter` is always there.

<img width="792" alt="image" src="https://user-images.githubusercontent.com/4665641/158981237-6fc1f23e-c909-46a2-adba-1355b494dc79.png">

**Reply by yusufkandemir** [reply]
@piperone
> I saw "General" and just chose that. Perhaps there should be a "CLI SPA-mode" category though, because my question doesn't relate to any of the existing ones (BEX, SSR, PWA, Capacitor or Electron). Should I move my question to one of the others?

[GitHub Discussions has a category limit](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-categories-for-discussions-in-your-repository#:~:text=Each%20repository%20can%20have%20up%20to%2010%20categories), so we can't create more categories. The category you've chosen is valid, but you didn't specify if you are using Quasar CLI with Vite or Quasar CLI with Webpack, and the version you are using. `Quasar 2.6.0` is just the UI package(_components, directives, utils, etc._), but your problem is not related to UI. All that was needed for this question was to tell if you are using `@quasar/app-vite` or `@quasar/app-webpack`, and the version.

> I don't think a reproduction repo is necessary

You can't imagine how many times I heard this and it proved to be wrong. I followed your repro steps, and couldn't reproduce the problem. I tested both Quasar CLI with Webpack and with Vite. I've used both Vuex and Pinia as the store provider. I couldn't reproduce the problem in any of them. So, please provide a repro, you can upload the project to GitHub, output of `quasar info` could be helpful along with the repro.

![image](https://user-images.githubusercontent.com/6266078/158980325-e04886d7-088f-46a4-8bdb-5059d81d18cb.png)

```
yarn create v1.22.17
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...

success Installed "create-quasar@1.0.9" with binaries:
      - create-quasar


 .d88888b.
d88P" "Y88b
888     888
888     888 888  888  8888b.  .d8888b   8888b.  888d888
888     888 888  888     "88b 88K          "88b 888P"
888 Y8b 888 888  888 .d888888 "Y8888b. .d888888 888
Y88b.Y8b88P Y88b 888 888  888      X88 888  888 888
 "Y888888"   "Y88888 "Y888888  88888P' "Y888888 888
       Y8b

✔ What would you like to build? › App with Quasar CLI, let's go!
✔ Project folder: … quasar-project
✔ Pick Quasar version: › Quasar v2 (Vue 3 | latest and greatest)
✔ Pick script type: › Javascript
✔ Pick Quasar App CLI variant: › Quasar App CLI with Webpack (stable)
✔ Package name: … quasar-project
✔ Project product name: (must start with letter if building mobile apps) … Quasar App
✔ Project description: … A Quasar Project
✔ Author: … Yusuf Kandemir <yusuf.kandemir@outlook.com.tr>
✔ Pick your CSS preprocessor: › Sass with SCSS syntax
✔ Check the features needed for your project: › ESLint, State Management (Pinia)
✔ Pick an ESLint preset: › Prettier

 Quasar • Generating files...

 - README.md
 - .editorconfig
 - .gitignore
 - .postcssrc.js
 - babel.config.js
 - jsconfig.json
 - package.json
 - quasar.config.js
 - .vscode/extensions.json
 - .vscode/settings.json
 - public/favicon.ico
 - src/App.vue
 - src/index.template.html
 - public/icons/favicon-128x128.png
 - public/icons/favicon-16x16.png
 - public/icons/favicon-32x32.png
 - public/icons/favicon-96x96.png
 - src/assets/quasar-logo-vertical.svg
 - src/boot/.gitkeep
 - src/components/EssentialLink.vue
 - src/layouts/MainLayout.vue
 - src/pages/ErrorNotFound.vue
 - src/pages/IndexPage.vue
 - src/router/index.js
 - src/router/routes.js
 - src/css/app.scss
 - src/css/quasar.variables.scss
 - .eslintignore
 - .eslintrc.js
 - src/stores/example-store.js
 - src/stores/index.js
 - src/stores/store-flag.d.ts

 Quasar •  SUCCESS  • The project has been scaffolded

✔ Install project dependencies? (recommended) › Yes, use yarn

yarn install v1.22.17
info No lockfile found.
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > @babel/eslint-parser@7.17.0" has unmet peer dependency "@babel/core@>=7.11.0".
warning " > eslint-webpack-plugin@3.1.1" has unmet peer dependency "webpack@^5.0.0".
[5/5] Building fresh packages...
success Saved lockfile.
Done in 30.76s.


To get started:

  cd quasar-project
  quasar dev # or: yarn quasar dev # or: npx quasar dev

Documentation can be found at: https://v2.quasar.dev

Quasar is relying on donations to evolve. We'd be very grateful if you can
read our manifest on "Why donations are important": https://v2.quasar.dev/why-donate
Donation campaign: https://donate.quasar.dev
Any amount is very welcomed.
If invoices are required, please first contact Razvan Stoenescu.

Please give us a star on Github if you appreciate our work:
  https://github.com/quasarframework/quasar

Enjoy! - Quasar Team
```

**Reply by piperone** [reply]
@rstoenescu I really appreciate you taking the time to investigate. Thank you!

Ultimately, the "issue" here was simple human error: In the installation process, I hadn't realized that the prompt where I choose store provider was a *selection*. I just highlighted Vuex and pressed enter. That, of course, meant that no store provider was installed. Now, I was coincidentally seeing the same error in my production app which I had just updated, so I assumed a connection between the two without there being one apparently.

I'm sorry for wasting everybody's time :-/ 

@yusufkandemir  Thanks for your feedback, Yusuf, and for taking the time to address my question. After asking in Discord and not getting anywhere, I still felt it was a bit overkill to create an issue for this, so I just figured I'd post a question here, hopefully bothering as few people as possible. I guess I was wrong there too; after reading your post, I really feel like I was being a nuisance. That was never my intention, of course, but if you want this to be a fruitful place of discussion between your users and you maintainers, being tactful goes a long way. Just my $0,02.

Thanks again!
